### PR TITLE
Spells & Source

### PR DESCRIPTION
**Pull request by MarcelloPaiva**
I added a `spell.ts` file to the schema folder and added the corresponding "spell" field to the `EntityData` in the `index.ts` file.
I also added a "source" field to the `EntityData` object.

Is there anything else I'm forgetting to add? Do I need to update my changes in the `schema.json file`?

### CODE REVIEW

File: src/schema/spell.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    /**
+     * The amount of damage the spell deals, if any.
+     * Takes an array of DamageData nodes, useful for spells that deal multiple damage types.
+     */
+    damage?: Array<DamageData>;
+}
+
+@component('spell')
+@requires('item')
```
Comment (drowrin) [inline src/schema/spell.ts]: did you mean 'description`?

Comment (MarcelloPaiva) [reply src/schema/spell.ts]: I think it got left there from me copying stuff but yeah description would work perfect to be required.

Comment (drowrin) [reply src/schema/spell.ts]: Gotcha. Otherwise looks great!

Comment (MarcelloPaiva) [reply src/schema/spell.ts]: fixed

**Review by drowrin — APPROVED** [review]

**Pull request merged**